lib/jnpr/junos/device.py

```diff
@@ -206,6 +206,119 @@ def port(self):
         """
         return self._port
 
+    # ------------------------------------------------------------------------
+    # property: master
+    # ------------------------------------------------------------------------
+
+    @property
+    def master(self):
+        """
+        The mastership state of the current Routing Engine.
+
+        The current Routing Engine is the RE to which the NETCONF session is
+        connected.
+
+        Note: This property is based on new-style fact gathering and the
+              value of currently cached facts. If there is a chance the
+              mastership state may have changed since the facts were cached,
+              then dev.facts_refresh() should be invoked prior to checking
+              this property. If old-style fact gathering is in use,
+              this property will return None.
+
+        :returns: True if the current RE is the master Routing Engine. False if
+                  the current RE is not the master Routing Engine. None if
+                  unable to determine the state of the current Routing Engine.
+        """
+        master = None
+
+        try:
+            # Make sure the 'current_re' fact has a value
+            if self.facts['current_re'] is not None:
+                # Typical master case
+                if 'master' in self.facts['current_re']:
+                    master = True
+                # Typical backup case
+                elif 'backup' in self.facts['current_re']:
+                    master = False
+                # Some single RE platforms don't have 'master' in the
+                # 'current_re' fact. It's best to check if it's a single RE
+                # platform based on the '2RE' fact, not the 'current_re' fact.
+                elif (self.facts['2RE'] is False and
+                      're0' in self.facts['current_re']):
+                    master = True
+                else:
+                    for re_state in self.facts['current_re']:
+                        # Multi-chassis case. A chassis master/backup, but
+                        # not the system master/backup.
+                        if '-backup' in re_state or '-master' in re_state:
+                            master = False
+                            break
+        except KeyError:
+            # Old style fact gathering is likely in use. Nothing we can do.
+            pass
```

Comment:
As per pep20 "Errors should never pass silently."

refer:https://www.python.org/dev/peps/pep-0020/